### PR TITLE
Use TestCommand in GivenThatIWantToMigrateTestApps

### PR DESCRIPTION
**Pull request by jonsequitur**
There's a small change here to use `TestCommand` in `GivenThatIWantToMigrateTestApps` in order to make it more consistent with other tests in this class.

An additional change here is to make the console output for `TestCommand` a bit more readable and refactor some of the related code to reduce duplication.

### CODE REVIEW

File: test/Microsoft.DotNet.Tools.Tests.Utilities/Commands/TestCommand.cs [hunk 1]
```diff
@@ -55,32 +55,37 @@ public virtual CommandResult Execute(string args = "")
...
 
             ResolveCommand(ref resolvedCommand, ref args);
-
-            var commandPath = Env.GetCommandPath(resolvedCommand, ".exe", ".cmd", "") ??
+            
+            Console.WriteLine($"[TestCommand] {WorkingDirectory}$ {resolvedCommand} {args}");
+            
+            if (captureOutput)
+            { 
+                var commandPath = Env.GetCommandPath(resolvedCommand, ".exe", ".cmd", "") ??
```
Comment (livarcocc) [inline test/Microsoft.DotNet.Tools.Tests.Utilities/Commands/TestCommand.cs]: What are we doing with commandpath now? It seems like it is not being used. Also, why only when captureOutput?

File: test/Microsoft.DotNet.Tools.Tests.Utilities/Commands/TestCommand.cs [hunk 2]
```diff
@@ -55,32 +55,37 @@ public virtual CommandResult Execute(string args = "")
...
         public async virtual Task<CommandResult> ExecuteAsync(string args = "")
         {
-            var resolvedCommand = _command;
-
-            ResolveCommand(ref resolvedCommand, ref args);
-
-            Console.WriteLine($"Executing - {resolvedCommand} {args} - {WorkingDirectoryInfo()}");
-            
-            return await ExecuteAsyncInternal(resolvedCommand, args);
+            return await ExecuteAsyncInternal(
```
Comment (krwq) [inline test/Microsoft.DotNet.Tools.Tests.Utilities/Commands/TestCommand.cs]: is there a point of having both ExecuteAsyncInternal and ExecuteAsync now?

File: test/dotnet-migrate.Tests/GivenThatIWantToMigrateTestApps.cs [hunk 3]
```diff
@@ -676,10 +676,10 @@ private void BuildProjectJson(string projectDirectory)
 
         private void MigrateProject(params string[] migrateArgs)
         {
-            var result =
-                MigrateCommand.Run(migrateArgs);
+            var result = new TestCommand("dotnet")
+                    .Execute($"migrate {string.Join(" ", migrateArgs)}");
 
-            result.Should().Be(0);
+            result.Should().Pass();
```
Comment (krwq) [inline test/dotnet-migrate.Tests/GivenThatIWantToMigrateTestApps.cs]: i think we should just chain-call Should().Pass() i don't think there is much value of having the variable

File: test/Microsoft.DotNet.Tools.Tests.Utilities/Commands/TestCommand.cs (hunk 1 above)
Comment (jonsequitur) [inline test/Microsoft.DotNet.Tools.Tests.Utilities/Commands/TestCommand.cs]: It wasn't being used. I wondered if someone was depending on a side effect so I didn't remove the call, but I can give that a try and see if any tests fail.

File: test/dotnet-migrate.Tests/GivenThatIWantToMigrateTestApps.cs (hunk 3 above)
Comment (jonsequitur) [inline test/dotnet-migrate.Tests/GivenThatIWantToMigrateTestApps.cs]: Agreed. I had it separated to write it to the console.

File: test/Microsoft.DotNet.Tools.Tests.Utilities/Commands/TestCommand.cs (hunk 1 above)
Comment (piotrpMSFT) [inline test/Microsoft.DotNet.Tools.Tests.Utilities/Commands/TestCommand.cs]: looks like dead code. Please remove while we have this context unboxed.

**Comment by piotrpMSFT** [comment]
@jonsequitur please do clean up the remaining unnecessary code as discussed by you and @livarcocc. Once finished, we'll merge this test cleanup change. @mattgertz FYI.

**Comment by jonsequitur** [comment]
This is a minor refactor and is better done on master.

**Pull request closed without merging**